### PR TITLE
minimal terraform templates

### PR DESCRIPTION
**Pull request by spohnan**
builds vpc, load balancer, ecs cluster with a single preloaded container and an optional bastion server

### CODE REVIEW

**Comment by jwilcoxen45** [comment]
Noted on this request!  Thank you!  I will review thorough on Monday morning.

File: pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  * `source env.sh`
+* Build stage environment infrastructure
+  * `terraform apply`
+
+### Result
+
+At the end of the deployment the following resources will have been provisioned.
+
+* VPC with two public and two private subnets
+* An optional bastion host which allows for console access to resources in private subnets
```
Comment (jwilcoxen45) [inline pipeline/README.md]: Nice use of bastion setup!

File: pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+* An ECS cluster provisioned within the private subnet
+* An Nginx container
+* An application load balancer
+
+EC2 -> Load Balancers link in the left-hand menu -> Select the ALB and copy the DNS hostname to access Nginx
+
+![architecture](diagrams/architecture.png)
```
Comment (jwilcoxen45) [inline pipeline/README.md]: Nice started diagrams.

File: pipeline/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+locals {
+  network_prefix = "10.1"
+  availability_zones = [ "${var.region}a", "${var.region}b" ]
+  resource_label = "${var.project}-${terraform.workspace}"
+}
+
+// ~~ Nginx container ~~~
+
+data "template_file" "nginx_task_definition" {
```
Comment (jwilcoxen45) [inline pipeline/terraform/main.tf]: Is this intended to be replaced with Java/spring container?  Or stuff java app within container?  Not picking up the reason for using nginx container.

Comment (spohnan) [reply pipeline/terraform/main.tf]: yes, nginx is just a placeholder that was used to verify that the networking and security groups were set up correctly. The application load balancer should be capable of whatever proxy capabilities will be needed and this initial container will be replaced with Jenkins, Tomcat etc. in future commits.

Comment (jwilcoxen45) [reply pipeline/terraform/main.tf]: Alright, then we will need an ECR too to push images.  I am sure that is in your thought process too.

**Pull request merged**